client/src/components/layers/BaseLayer.ts

```diff
@@ -0,0 +1,100 @@
+/*eslint class-methods-use-this: "off"*/
+import Track, { Feature, ConfidencePair, TrackId } from '@/lib/track';
+import { FrameDataTrack, CoordinateList } from '@/components/layers/LayerTypes';
+
+import { boundToGeojson } from '@/utils';
+import { StateStyles } from '@/use/useStyling';
+import geo from 'geojs';
+
+
+interface LayerStyle {
+    strokeWidth?: (number | ((a: any, b: any, data: any) => number));
+    strokeOpacity?: (number | ((a: any, b: any, data: any) => number));
+    strokeColor?: (string | ((a: any, b: any, data: any) => string));
+}
+
+interface BaseLayerParams {
+    frameData?: FrameDataTrack;
+    annotator?: any;
+    stateStyling?: StateStyles;
+    typeColorMap?: (t: string) => string;
+    signals?: Record<string, any>;
+}
+
+export default class BaseLayer {
+    frameData?: FrameDataTrack[];
+
+    formattedData: any;
+
+    annotator: any;
+
+    stateStyling: StateStyles;
+
+    style: LayerStyle;
+
+    typeColorMap: (t: string) => string;
+
+    redrawSignalers: any[];
+
+    featureLayer: any;
+
+    signals: any;
+
+    constructor({
+      frameData,
+      annotator,
+      stateStyling,
+      typeColorMap,
+      signals = {
+        annotationClicked: () => null, annotationRightClicked: () => null,
+      },
```

Comment:
export default class BaseLayer extends Vue would allow you to have all the conveniences of `$emit` and `$on` witthout adding any real overhead.

We would have to explain that caution should be used when deciding to use other features of the vue instance in the future, but that's simple enough.

Reply:
additionally before I forget, these default functions should be `AnnotationLayer.ts` not the `BaseLayer.ts`